cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/realtime/sources/JmsSource.java

```diff
@@ -44,15 +44,18 @@
 public class JmsSource extends RealtimeSource<String> {
   private static final Logger LOG = LoggerFactory.getLogger(JmsSource.class);
 
-  private static final long JMS_CONSUMER_TIMEOUT_MS = 30000;
+  private static final long JMS_CONSUMER_TIMEOUT_MS = 2000;
```

Comment:
Not right now. The idea since we are polling the data anyway, we just control the timeout from JMS server to let the caller poll again if no messages or timeout.